app/clients/zendesk_sell_client.py

```diff
@@ -0,0 +1,94 @@
+import json
+import requests
+
+from typing import Dict, List, Union
+
+from flask import current_app
+
+from app.authentication.bearer_auth import BearerAuth
+from app.user.contact_request import ContactRequest
+
+__all__ = [
+    'ZenDeskSellClient'
+]
+
+
+class ZenDeskSellClient(object):
+
+    def _generate_lead_data(self) -> Dict[str, Union[str, List[str], Dict]]:
+
+        # The API and field definitions are defined here: https://developers.getbase.com/docs/rest/reference/leads
+
+        # validation based upon api mandatory fields
+        assert len(self.contact.name) or len(self.contact.department_org_name), 'Name or Org are mandatory field'
+
+        recipients = {
+            'internal': 'Colleagues within your department (internal)',
+            'external': 'Partners from other organizations (external)',
+            'public': 'Public'
+        }
+        # FIXME: consider migrating to pypi/nameparser for proper name parsing to handle cases like:
+        # 'Dr. Juan Q. Xavier de la Vega III (Doc Vega)'
+        name_tokenised = self.contact.name.split()
+        return {
+            'data': {
+                'last_name': name_tokenised[-1],
+                'first_name': " ".join(name_tokenised[:-1]) if len(name_tokenised) > 1 else '',
+                'organization_name': self.contact.department_org_name,
+                'email': self.contact.email_address,
+                'description': f'Program: {self.contact.program_service_name}\n{self.contact.main_use_case}: '
+                               f'{self.contact.main_use_case_details}',
+                'tags': [self.contact.support_type, self.contact.language],
```

Comment:
Did you get with folks in charge of the CRM for these tags? Would language be better as a custom field?

Reply:
Amy is talking with Leanne as we speak regarding en/fr determination.
This is a work in progress at this point. There are two hard coded fields which overload for EN/FR but also indicating EN/FR for signing up to the mailing list. Amy also has proposed potential changes within the admin intake form to explicitly ask for language preference regardless if they are viewing the UI in EN or FR. As long as we have this tag here now, we can start ironing out those details in a subsequent change once we know the design better